extension/src/stats_agg.rs

```diff
@@ -0,0 +1,820 @@
+                (None, None) => Some(StatsSummary2D::from_internal(InternalStatsSummary2D::new()).into()), // return an empty one from the trans function because otherwise it breaks in the window context
+                (Some(state), None) => Some(state),
+                (None, Some(val)) => {
+                    let mut s = InternalStatsSummary2D::new();
+                    s.accum(val).unwrap();
+                    Some(StatsSummary2D::from_internal(s).into())
+                },
+                (Some(mut state), Some(val)) => {
+                    let mut s: InternalStatsSummary2D = state.to_internal(); 
+                    s.accum(val).unwrap();
+                    *state = StatsSummary2D::from_internal(s);
+                    Some(state)
+                },
+            }
+        })
+    }
+}
+
+
+#[pg_extern(schema = "timescale_analytics_experimental")]
+pub fn stats1d_inv_trans<'s>(
+    state: Option<Internal<StatsSummary1D<'s>>>,
+    val: Option<f64>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal<StatsSummary1D<'s>>> {
+    unsafe {
+        in_aggregate_context(fcinfo, || {
+            match (state, val) {
+                (None, _) => panic!("Inverse function should never be called with NULL state"),
+                (Some(state), None) => Some(state),
+                (Some(state), Some(val)) => {
+                    let s: InternalStatsSummary1D = state.to_internal(); 
+                    let s = s.remove(val);
+                    match s {
+                        None => None,
+                        Some(s) => Some(StatsSummary1D::from_internal(s).into())
+                    }
+                },
+            }
+        })
+    }
+}
+
+#[pg_extern(schema = "timescale_analytics_experimental")]
+pub fn stats2d_inv_trans<'s>(
+    state: Option<Internal<StatsSummary2D<'s>>>,
+    y: Option<f64>,
+    x: Option<f64>,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal<StatsSummary2D<'s>>> {
+    unsafe {
+        in_aggregate_context(fcinfo, || {
+            let val: Option<XYPair> = match (y, x) {
+                (None, _) => None,
+                (_, None) => None,
+                (Some(y), Some(x)) => Some(XYPair{y, x})
+            };
+            match (state, val) {
+                (None, _) => panic!("Inverse function should never be called with NULL state"),
+                (Some(state), None) => Some(state),
+                (Some(state), Some(val)) => {
+                    let s: InternalStatsSummary2D = state.to_internal(); 
+                    let s = s.remove(val);
+                    match s {
+                        None => None,
+                        Some(s) => Some(StatsSummary2D::from_internal(s).into())
+                    }
+                },
+            }
+        })
+    }
+}
+
+
+#[pg_extern(schema = "timescale_analytics_experimental")]
+pub fn stats1d_summary_trans<'s, 'v>(
+    state: Option<Internal<StatsSummary1D<'s>>>,
```

Comment:
It seems like the logic of all of these pg aggregate functions would be streamlined if the internal representation was a `StatsSummary1DInternal`.  This would require a separate trans function for the rollup, but still seems like it might be cleaner.